google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/flexible/AppEngineFlexibleDeploymentEditor.java

```diff
@@ -147,116 +129,92 @@ protected void textChanged(DocumentEvent event) {
         }
     );
 
-    appYamlOverrideCheckBox.addActionListener(event -> {
-      boolean isAppYamlOverrideSelected = ((JCheckBox) event.getSource()).isSelected();
-      appYamlTextField.setVisible(isAppYamlOverrideSelected);
-
-      setModuleControlsEnabled(
-          !(isAppYamlOverrideSelected && dockerDirectoryOverrideCheckBox.isSelected()));
-      updateServiceName();
+    resetAppYamlConfigSelection(project);
+    appYamlCombobox.addItemListener(event -> {
       toggleDockerfileSection();
+      resetRuntimeDisplay();
     });
-
-    dockerDirectoryOverrideCheckBox.addActionListener(
-        event -> {
-          boolean isDockerfileOverrideSelected = ((JCheckBox) event.getSource()).isSelected();
-          dockerDirectoryTextField.setEnabled(isDockerfileOverrideSelected);
-          if (isDockerfileOverrideSelected) {
-            if (dockerfileOverride.isEmpty()) {
-              dockerfileOverride = dockerDirectoryTextField.getText();
-            }
-            dockerDirectoryTextField.setText(dockerfileOverride);
-          } else {
-            dockerfileOverride = dockerDirectoryTextField.getText();
-            dockerDirectoryTextField.setText(getDockerDirectoryPath());
-          }
-
-          setModuleControlsEnabled(
-              !(isDockerfileOverrideSelected && appYamlOverrideCheckBox.isSelected()));
-        }
-    );
-
-    appYamlTextField.getTextField().getDocument().addDocumentListener(new DocumentAdapter() {
-      @Override
-      protected void textChanged(DocumentEvent event) {
-        updateServiceName();
-        toggleDockerfileSection();
-      }
-    });
-
-    resetModuleConfigSelection(project);
-    modulesWithFlexFacetComboBox.addItemListener(event -> toggleDockerfileSection());
-    modulesWithFlexFacetComboBox.setRenderer(new ListCellRendererWrapper<Module>() {
+    appYamlCombobox.setRenderer(new ListCellRendererWrapper<AppEngineFlexibleFacet>() {
       @Override
-      public void customize(JList list, Module value, int index, boolean selected,
+      public void customize(JList list, AppEngineFlexibleFacet facet, int index, boolean selected,
```

Comment:
nit: Use the Google Java formatter to fix this, but the params should either be altogether on the next line (indented by 4 spaces), or if they don't fit like that either, all on separate lines indented by 4 spaces.